core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.controller
+
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+
+import spray.client.pipelining._
+import spray.http._
+import spray.http.HttpMethods._
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import spray.routing.Directives
+import whisk.common.TransactionId
+import whisk.core.controller.actions.PostActionActivation
+import whisk.core.database._
+import whisk.core.entity._
+import whisk.core.entity.types._
+import whisk.http.ErrorResponse.terminate
+
+trait WhiskMetaApi extends Directives with PostActionActivation {
+    services: WhiskServices =>
+
+    /** API path and version for posting activations directly through the host. */
+    val apipath: String
+    val apiversion: String
+
+    /** Store for identities. */
+    protected val authStore: AuthStore
+
+    /** The route prefix e.g., /experimental/package-name. */
+    protected val routePrefix = pathPrefix("experimental")
+
+    /** The name and apikey of the system namespace. */
+    protected val systemId = "whisk.system"
+    protected lazy val systemKey = WhiskAuth.get(authStore, Subject(systemId), false)(TransactionId.controller)
+
+    /** Allowed verbs. */
+    private lazy val allowedOperations = get | delete | post
+
+    /** Extracts the HTTP method and query params. */
+    private val requestMethodAndParams = {
+        extract(ctx => (ctx.request.method, ctx.request.message.uri.query.toMap))
+    }
+
+    def routes(user: Identity)(implicit transid: TransactionId) = {
+        (routePrefix & pathPrefix(EntityName.REGEX.r) & allowedOperations) { s =>
```

Comment:
As discussed in person:
It will be worthwhile getting the rest of the unmatched path an pass it to the actions as `path` or `entity` parameter. Makes it possible to create a more RESTful API with this approach.